### PR TITLE
server: fix cpu loop on SIGPIPE

### LINKED ISSUE #6543: exit of vscode remote containter leaves processes with high cpu in container
- VSCode Version: 1.66.0
- Local OS Version: Ubuntu 20.04
- Remote OS Version: CentOS 7.6
- Remote Extension/Connection Type:  Docker
- Logs:

Steps to Reproduce:

1. Run vscode remote container
2. exit vscode 

 
Does this issue occur when you try this locally?:  No
Does this issue occur when you try this locally and all extensions are disabled?: No
 

Full cmd line of high cpu process:
```
/root/.vscode-server/bin/e18005f0f1b33c29e81d732535d8c0e47cafb0b5/node /root/.vscode-server/bin/e18005f0f1b33c29e81d732535d8c0e47cafb0b5/out/server-main.js --log debug --force-disable-user-env --server-data-dir /root/.vscode-server --use-host-proxy --accept-server-license-terms --host 127.0.0.1 --port 0 --connection-token-file /root/.vscode-server/data/Machine/.connection-token-e18005f0f1b33c29e81d732535d8c0e47cafb0b5 --extensions-download-dir /root/.vscode-server/extensionsCache --start-server --disable-websocket-compression
```
There are 2 such processes, each using one core of CPU to its full. One with 2M/s memory increase, another with 0.2M/s memory usage increase. I only noticed this after today 1.66.0 vscode update. Not sure if it occurred before.

### PR DESCRIPTION
**Pull request by connor4312**
- One missed uncaughtException site was causing a loop (maybe we want to handle this error on onUnexpectedError instead?)
- The SIGPIPE listener itself was causing a loop, as well. Only try to log once.

Fixes https://github.com/microsoft/vscode-remote-release/issues/6543

### CODE REVIEW

File: src/vs/platform/telemetry/node/errorTelemetry.ts [hunk 1]
```diff
@@ -44,6 +44,14 @@ export default class ErrorTelemetry extends BaseErrorTelemetry {
 
 		// Print a console message when an exception isn't handled.
 		process.on('uncaughtException', (err: Error) => {
+			// See https://github.com/microsoft/vscode-remote-release/issues/6481
+			// In some circumstances, console.error will throw an asynchronous error. This asynchronous error
+			// will end up here, and then it will be logged again, thus creating an endless asynchronous loop.
+			// Here we try to break the loop by ignoring EPIPE errors that include our own unexpected error handler in the stack.
+			if (err && (err as any).code === 'EPIPE' && (err as any).syscall === 'write') {
```
Comment (bpasero) [inline src/vs/platform/telemetry/node/errorTelemetry.ts:51]: Fyi: `ErrnoException`

**Comment by bpasero** [comment]
Do we need a similar protection here?

https://github.com/microsoft/vscode/blob/f370b736aa813da9a6f6b82a48bfc0a0d19b70f6/src/bootstrap.js#L34-L36

**Review by AaronWashington — DISMISSED** [review]

**Comment by connor4312** [comment]
That code path was not hit in my testing, I believe due to 

https://github.com/microsoft/vscode/blob/844848d88466175870063e8bccfeced4ca41c7a0/src/server-main.js#L268

Though this seems like a fragile scenario...

**Comment by bpasero** [comment]
@connor4312 if we do have evidence that `console.error` can cause `SIGPIPE`, I would suggest to add the same protection you introduce via `didLogAboutSIGPIPE` also here:

https://github.com/microsoft/vscode/blob/4c0cecf300334511c6ef64f40ba584dcbe3a03de/src/bootstrap.js#L34-L36

**Review by sophacola — APPROVED** [review]

**Review by bpasero — APPROVED** [review]

**Comment by alexdima** [comment]
Catching up with my notifications. Thank you @connor4312 for tracking this down!

**Pull request merged**